### PR TITLE
Composable tasks to perform complex actions

### PR DESCRIPTION
**Pull request by CostinCarabas**

### CODE REVIEW

File: contracts/proxy-actions/src/structs.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+                    let token_out = TokenIdentifier::from(args.get(0).clone_value());
+                    let min_amount_out = BigUint::from(args.get(1).clone_value());
+                    let token_payment = payment_to_next_task
+                        .token_identifier
+                        .clone()
+                        .into_esdt_option();
+                    if token_payment.is_none() {
+                        return;
+                    }
+                    self.pair_contract_proxy(dest_addr)
```
Comment (sasurobert) [inline contracts/proxy-actions/src/structs.rs:95]: I think it would be better to call the router ? it can do all kinds of swaps.

Comment (sasurobert) [reply contracts/proxy-actions/src/structs.rs:95]: And you would know the router address by default. So it is a setting in the contract. The same is for wrap contract address.

Comment (sasurobert) [reply contracts/proxy-actions/src/structs.rs:95]: pair contract addresses can be written in storage.

**Comment by CostinCarabas** [comment]
Will continue this PR here [1]

[1] https://github.com/multiversx/mx-exchange-tools-sc/pull/44

**Pull request closed without merging**